Assets/MixedRealityToolkit.SDK/Features/UX/Interactable/Scripts/Interactable.cs

```diff
@@ -221,149 +262,258 @@ public List<InteractableEvent> InteractableEvents
+        public virtual bool HasPress
+        {
+            get { return GetStateValue(InteractableStates.InteractableStateEnum.Pressed) > 0; }
+            set { SetState(InteractableStates.InteractableStateEnum.Pressed, value); }
+        }
 
-        // advanced button states from InteractableStates.InteractableStateEnum
         /// <summary>
-        /// Has focus, finger up - custom: not set by Interactable
+        /// Targeted means the item has focus and finger is up
+        /// Currently not controlled by Interactable directly
         /// </summary>
-        public bool IsTargeted { get; private set; }
+        public virtual bool IsTargeted
+        {
+            get { return GetStateValue(InteractableStates.InteractableStateEnum.Targeted) > 0; }
+            set { SetState(InteractableStates.InteractableStateEnum.Targeted, value); }
+        }
 
         /// <summary>
-        /// No focus, finger is up - custom: not set by Interactable
+        /// State that corresponds to no focus,and finger is up.
+        /// Currently not controlled by Interactable directly
         /// </summary>
-        public bool IsInteractive { get; private set; }
+        public virtual bool IsInteractive
+        {
+            get { return GetStateValue(InteractableStates.InteractableStateEnum.Interactive) > 0; }
+            set { SetState(InteractableStates.InteractableStateEnum.Interactive, value); }
+        }
 
         /// <summary>
-        /// Has focus, finger down - custom: not set by Interactable
+        /// State that corresponds to has focus,and finger down.
+        /// Currently not controlled by Interactable directly
         /// </summary>
-        public bool HasObservationTargeted { get; private set; }
+        public virtual bool HasObservationTargeted
+        {
+            get { return GetStateValue(InteractableStates.InteractableStateEnum.ObservationTargeted) > 0; }
+            set { SetState(InteractableStates.InteractableStateEnum.ObservationTargeted, value); }
+        }
 
         /// <summary>
-        /// No focus, finger down - custom: not set by Interactable
+        /// State that corresponds to no focus,and finger is down.
+        /// Currently not controlled by Interactable directly
         /// </summary>
-        public bool HasObservation { get; private set; }
+        public virtual bool HasObservation
+        {
+            get { return GetStateValue(InteractableStates.InteractableStateEnum.Observation) > 0; }
+            set { SetState(InteractableStates.InteractableStateEnum.Observation, value); }
+        }
 
         /// <summary>
         /// The Interactable has been clicked
         /// </summary>
-        public bool IsVisited { get; private set; }
+        public virtual bool IsVisited
+        {
+            get { return GetStateValue(InteractableStates.InteractableStateEnum.Visited) > 0; }
+            set { SetState(InteractableStates.InteractableStateEnum.Visited, value); }
+        }
```

Comment:
when does this get reset?

Reply:
Only via ResetAllStates() which is only called if a new States type is assigned in SetupStates(). 

In general practice, it should be reset because once I visited, it's just true for life.

Reply:
Yeah. It might be a good idea  to reset it after certain time period ?

---
In reply to: [329205597](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/6104#discussion_r329205597) [](ancestors = 329205597)

Reply:
Sorry I mistyped my original comment. It should NOT be reset. The state tracks whether the button has been every clicked. BY definition, it needs to stay true unless I am changing the state manager.